File: webApp/src/types/animal.ts
export const ANIMAL_STATUSES = ['MISSING', 'FOUND', 'CLOSED'] as const;
export type AnimalStatus = (typeof ANIMAL_STATUSES)[number];

export const ANIMAL_SEXES = ['MALE', 'FEMALE', 'UNKNOWN'] as const;
export type AnimalSex = (typeof ANIMAL_SEXES)[number];

export const ANIMAL_SPECIES = ['DOG', 'CAT', 'BIRD', 'RABBIT', 'OTHER'] as const;
export type AnimalSpecies = (typeof ANIMAL_SPECIES)[number];

export const ANIMAL_STATUS_BADGE_COLORS: Record<AnimalStatus, string> = {
  MISSING: '#FF0000',
  FOUND: '#0074FF',
  CLOSED: '#93A2B4'
};

export interface Animal {
  // Required fields
  id: string; // UUID string
  photoUrl: string; // URL to pet photo
  status: AnimalStatus; // MISSING, FOUND, or CLOSED
  lastSeenDate: string; // ISO 8601 date (YYYY-MM-DD)
  species: AnimalSpecies; // Pet species (DOG, CAT, BIRD, RABBIT, OTHER)
  sex: AnimalSex; // MALE, FEMALE, or UNKNOWN

  // Basic information (nullable)
  petName: string | null; // Pet's name
  breed: string | null; // Pet breed
  description: string | null; // Additional description text

  // Location (nullable)
  locationLatitude: number | null; // Latitude coordinate
  locationLongitude: number | null; // Longitude coordinate

  // Contact information (nullable, at least one should be present)
  phone: string | null; // Phone number (may be masked by backend)
  email: string | null; // Email address

  // Optional detailed fields (nullable)
  microchipNumber: string | null; // Raw microchip number (formatted in component)
  age: number | null; // Approximate age
  reward: string | null; // Reward amount (displayed as-is)
  createdAt: string | null; // ISO 8601 timestamp
  updatedAt: string | null; // ISO 8601 timestamp
}

File: composeApp/src/androidMain/kotlin/com/intive/aifirst/petspot/composeapp/domain/models/Animal.kt
package com.intive.aifirst.petspot.composeapp.domain.models

/**
 * Represents an animal in the PetSpot system.
 * Contains all information needed for list display and detail views.
 * Shared across all platforms via Kotlin Multiplatform.
 *
 * @property id Unique identifier (UUID or database ID)
 * @property name Name of the animal (e.g., "Buddy", "Mittens")
 * @property photoUrl URL or placeholder identifier for animal photo
 * @property location Geographic location with radius for search area
 * @property species Animal species (Dog, Cat, Bird, etc.)
 * @property breed Specific breed name (e.g., "Maine Coon", "German Shepherd")
 * @property gender Biological sex (Male, Female, Unknown)
 * @property status Current status (Active, Found, Closed)
 * @property lastSeenDate Date when animal was last seen (for Active status) or found (for Found status)
 * @property description Detailed text description (visible on web, truncated on mobile)
 * @property email Contact email of the person who reported/owns the animal (optional)
 * @property phone Contact phone number of the person who reported/owns the animal (optional)
 * @property microchipNumber Microchip number for pet identification (optional)
 * @property rewardAmount Reward amount as string, displayed as-is (optional)
 * @property approximateAge Approximate age description (optional)
 */
data class Animal(
    val id: String,
    val name: String,
    // URL or placeholder identifier
    val photoUrl: String,
    val location: Location,
    val species: AnimalSpecies,
    val breed: String,
    val gender: AnimalGender,
    val status: AnimalStatus,
    // Format: DD/MM/YYYY (as specified in spec)
    val lastSeenDate: String,
    val description: String,
    // Optional contact email
    val email: String?,
    // Optional contact phone
    val phone: String?,
    // Optional microchip number (format: 000-000-000-000)
    val microchipNumber: String? = null,
    // Optional reward amount (displayed as-is)
    val rewardAmount: String? = null,
    // Optional approximate age description
    val approximateAge: String? = null,
)
